File: tests/parsers/samples/code/zig/simple.zig
const std = @import("std");
const complex = @import("complex.zig");

pub const Metrics = struct {
    baseline: f64,
    rms: f64,
};

pub fn runSimple() Metrics {
    const analyzer = complex.buildDefault();
    const values = [_]f64{ 1, 2, 3 };
    return .{
        .baseline = analyzer.distance(10, 3),
        .rms = analyzer.rootMeanSquare(&values),
    };
}

pub fn summarize() Metrics {
    return runSimple();
}

File: tests/parsers/samples/code/rs/simple.rs
use crate::complex::{build_default, ComplexMetrics};

pub struct Metrics {
    pub baseline: f64,
    pub rms: f64,
}

pub fn run_simple() -> Metrics {
    let analyzer = build_default();
    Metrics {
        baseline: analyzer.distance(10, 3),
        rms: analyzer.root_mean_square(&[1.0, 2.0, 3.0]),
    }
}

pub fn summarize() -> String {
    let metrics = run_simple();
    format!("baseline={:.2}, rms={:.2}", metrics.baseline, metrics.rms)
}
